### PR TITLE
[fix](mtmv) Fix compensate union all wrongly when query rewrite by materialized view

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

This is brought by https://github.com/apache/doris/pull/36056

Not all query after rewritten successfully can compensate union all
Such as:
mv def sql is as following, partition column is a
```sql
select a, b, count(*) from t1 group by a, b
```
Query is as following:
```sq
select count(*) from t1
```
the result is
+----------+
| count(*) |
+----------+
|       24 |
+----------+

after rewritten by materialized view successfully
If mv part partition is invalid, can not compensate union all, because result is wrong after
compensate union all.

+----------+
| count(*) |
+----------+
|       24 |
|       3 |
+----------+

This pr fix this.

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java [hunk 1]
```diff
@@ -324,6 +326,46 @@ protected Expression tryRewriteExpression(StructInfo queryStructInfo, Expression
...
+        }
+        String relatedCol = mtmv.getMvPartitionInfo().getRelatedCol();
+        boolean canUnionRewrite = false;
+        // Check the query plan group by expression contains partition col or not
+        List<? extends Expression> groupByShuttledExpressions =
+                ExpressionUtils.shuttleExpressionWithLineage(groupByExpressions, queryPlan, new BitSet());
+        for (Expression expression : groupByShuttledExpressions) {
+            canUnionRewrite = !expression.collectToSet(expr -> expr instanceof SlotReference
+                    && ((SlotReference) expr).isColumnFromTable()
+                    && ((SlotReference) expr).getColumn().get().getName().equals(relatedCol)).isEmpty();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java:361]: check table name too

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**